README.md

```diff
@@ -30,36 +30,40 @@ The ClusterGroupUpgrade controller is designed to perform an upgrade to a group
 
-A full table of the conditions and their appropriate reasons and types is:
+A full table of the conditions and their appropriate reasons and types are:
 
   | Type | Status| Reason| Message |
   |------|-------|---------|--------|
-  ClustersSelected| True| Completed||
-  | |False | NotFound/NotPresent| `comment: mention the cluster list that is not part of the manangedCluster, must block the upgrade`|
-  Validated | True | Completed||
-  | | False | MissingManagedPolicy| `comment: must block the upgrade`|
-  | | False | InvalidPlatformImage| `comment: must block the upgrade`| 
-  `BackupSucceeded` | True | Completed | Backup is completed |
-  | | True | PartiallyDone | Failed to create backups for some clusters |
-  | | False | InProgress | Backup is in progress |
-  | | False | Failed | Backup failed for all the clusters |
-  `PrecacheSpecValid` | True | PrecacheSpecIsWellFormed/AsExpected | Pre-caching spec is valid and consistent|
+  `ClustersSelected`| True| Completed| All selected clusters are valid for upgrade| 
+  | |False | NotFound/NotPresent| Unable to select clusters: error message |
+  `Validated` | True | Completed| Completed validation |
+  | | False | NotAllManagedPoliciesExist| Missing managed policies: policyList,  invalid managed policies: policyList |
+  | | False | InvalidPlatformImage | Precache spec is incomplete |
+  `PrecacheSpecValid` | True | PrecacheSpecIsWellFormed | Precaching spec is valid and consistent |
+  | | False | InvalidPlatformImage| Precaching spec is incomplete |
   | | False | NotAllManagedPoliciesExist| The ClusterGroupUpgrade CR has managed policies that are missing|
-  `PrecachingSucceeded` | True | Completed | Precaching is completed |
-  | | True | PartiallyDone | Failed to create preCaching for some clusters | 
-  | | False | InProgress | Precaching is in progress | 
-  | | False | Failed | PreCaching failed for all the clusters |
-  Progressing| True | InProgress||
-  | | False | Completed | |
+  `PrecachingSucceeded` | True | Completed | Precaching is completed for all clusters|
+  | | True | PartiallyDone | Precaching failed for x clusters | 
+  | | False | InProgress | Precaching is not completed | 
+  | | False | InProgress | Precaching is in progress for x clusters | 
+  | | False | Failed | Precaching failed for all clusters |
+  `BackupSucceeded` | True | Completed | Backup is completed for all clusters|
+  | | True | PartiallyDone | Backup failed for x clusters |
+   | | False | NotEnabled | Backup will start when enable is true |
+  | | False | InProgress | Backup is in progress for x clusters|
+  | | False | Failed | Backup failed for all the clusters |
+  `Progressing`| True | InProgress| The ClusterGroupUpgrade CR has upgrade policies that are still non compliant|
+  | | False | Completed | The ClusterGroupUpgrade CR has all clusters compliant with all the managed policies |
+  | | False | NotStarted | The Cluster backup is in progress |
```

Comment:
In Steven's PR (@fontivan), we are streamlining the cluster selection. We stop re-doing cluster selection/policy validation when progressing becomes true. With this change, the condition for that part would become more complicated. Do you think it would make more sense to consider backup part of progressing (as we can't redo cluster selection at that point)?

Reply:
The problem is if precaching and backup are set to true, and if couple of clusters fails, we have filter out the successful clusters and rebuild/update cluster selection and remediation plan. I don't know why https://github.com/openshift-kni/cluster-group-upgrades-operator/pull/348 is marked as a bug, for me it is more of an improvement. But the easier way is to if Progressing is set to true (it is only true when there is no blocking CR, missing CR, precaching and backup is done or failed), we can just check progressing == true and don't do cluster selection/policy validation.